mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala

```diff
@@ -617,8 +617,27 @@ private[spark] object BLAS extends Serializable with Logging {
       x: SparseVector,
       beta: Double,
       y: DenseVector): Unit = {
-    val xValues = x.values
-    val xIndices = x.indices
+    
+    def isSorted(l:Array[Int]): Boolean = {
+      var index = 1
+      while(index < l.length){
+        val prevElement = l(index - 1)
+        val currentElement = l(index)
+        
+        if(prevElement > currentElement) return false
+        index += 1
+      }
+      return true
+    }
+    
+    val (xValues,xIndices) = isSorted(x.indices) match{
```

Comment:
match is overkilll for a boolean. Just use an `if`. We should avoid the style that omits member dereference "." here. Also you can use `unzip` I think

```
val (xValues, xIndices) =
  if (isSorted(x.indices)) {
    (x.values, x.indices)
  } else {
    x.values.zip(x.indices).sortBy(_._2).unzip
  }
```

The `def isSorted` probably should just be inlined, even, or made non-local